pkg/apis/build/v1alpha1/image_builds.go

```diff
@@ -23,7 +23,14 @@ const (
 	BuildReasonBuildpack  = "BUILDPACK"
 )
 
-func (im *Image) buildNeeded(lastBuild *Build, sourceResolver *SourceResolver, builder *Builder) ([]string, bool) {
+type AbstractBuilder interface {
+	metav1.ObjectMetaAccessor
+	ImageRef() *BuilderImage
```

Comment:
It seems like we always dereference this. Perhaps, we should change remove the pointer to BuilderImage?

Reply:
Done